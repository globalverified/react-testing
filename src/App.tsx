import React from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      link
      <header>
        <h1 data-testid="heading">D14-Squard-6</h1>
        <h3 data-testid="subheading">
          <a href="#" data-testid="link">Sub-Heading</a>
          <a href="#" data-testid="link2">Sub-Heading2</a>
        </h3>
      </header>
      <section>
        <a className="link-self"
          href="https://google.com"
          target="_self"
          rel="external" data-testid="link">This link</a> will open in same tab and
        <a className="link-newtab"
          href="https://google.com"
          target="_blank"
          rel="external" data-testid="link">this link</a> will open in new tab.
      </section>
    </div>
  );
}

export default App;
