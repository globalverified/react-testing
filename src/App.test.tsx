import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import App from './App';
/* eslint-disable import/no-extraneous-dependencies */
import toJson from 'enzyme-to-json'; 

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({ adapter: new Adapter() });

/*
Group the test scenario
*/
describe('Full Testing of Entire App Component', () => {
  const component = shallow(<App />);

  it('it should run App Component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  test('it should have one heading', () => {
    expect(component.exists('h1'));// please add matcher call like .toBe(), .toEqual(), also try to add one more h1 tag and see if this work
  });

  test('heading with text "D14-Squard-6', () => {
    expect(component.find('h1').text()).toEqual('D14-Squard-6'); 
  });

  test('it should have sub heading', () => {
    expect(component.exists('h3')); // please add matcher call like .toBe(), .toEqual(), also try to add one more h3 tag and see if this work
  });

  test('it should have 1 link in sub heading', () => {
    const link = component.find('h3').exists('a');
    expect(link); // please add matcher call like .toBe(), .toEqual(), also try to add one more anchor tag and see if this work
  });

  test('it should have 3 only links in entire app', () => {
    // expect(component.find('a').length == 3); // this test never fail - please add matcher call like .toBe(), .toEqual(), also try to add one more h2 tag and see if this work
    expect(component.find('a').length).toEqual(3);
  });

  test('it should have 2 links in body', () => {
    expect(component.find('section').find('a').length).toEqual(2);
  });

  test('first link of body should open in same tab', () => {
    const firstLink = component.find('section').find('.link-self');  
    expect(firstLink.prop('target')).toEqual('_self');  //validate first link of body
  });

  test('first link of body should open in new tab', () => {
    const secondlink = component.find('section').find('.link-newtab'); 
    expect(secondlink.prop('target')).toEqual('_blank'); //validate second open in new
  });

});

/*
Write a test cases in jest to cover below scenarios:
it should have one heading
with text "D14-Squard-6"
it should have 3 only links
it should have 2 links in body
it should have 1 link in sub heading
validate that first link of body should open in same tab 
and second open in new
 */
