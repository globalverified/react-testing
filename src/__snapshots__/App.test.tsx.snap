// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Testing of Entire App Component it should run App Component 1`] = `
<div
  className="App"
>
  link
  <header>
    <h1
      data-testid="heading"
    >
      D14-Squard-6
    </h1>
    <h3
      data-testid="subheading"
    >
      <a
        data-testid="link"
        href="#"
      >
        Sub-Heading
      </a>
    </h3>
  </header>
  <section>
    <a
      className="link-self"
      data-testid="link"
      href="https://google.com"
      rel="external"
      target="_self"
    >
      This link
    </a>
     will open in same tab and
    <a
      className="link-newtab"
      data-testid="link"
      href="https://google.com"
      rel="external"
      target="_blank"
    >
      this link
    </a>
     will open in new tab.
  </section>
</div>
`;
